datafusion/core/src/datasource/listing/table.rs

```diff
@@ -1828,7 +1862,7 @@ mod tests {
         )
         .await
         .expect_err("Example should fail!");
-        assert_eq!("Error during planning: zstd compression requires specifying a level such as zstd(4)", format!("{e}"));
+        assert_eq!("Invalid Option: zstd compression requires specifying a level such as zstd(4)", format!("{e}"));
```

Comment:
This example I think shows why I wanted to create a new error type. "Error during planning" I think would be more confusing to an end user than just a straightforward "Invalid Option" error.